cmd/cloud-controller-manager/controller-manager.go

```diff
@@ -17,32 +17,66 @@ limitations under the License.
 // The external controller manager is responsible for running controller loops that
 // are cloud provider dependent. It uses the API to listen to new events on resources.
 
+// This file should be written by each cloud provider.
+// The current file demonstrate how other cloud provider should leverage CCM and it uses fake parameters. Please modify for your own use.
+
 package main
 
 import (
 	"math/rand"
 	"os"
 	"time"
 
+	"github.com/spf13/pflag"
+	cloudprovider "k8s.io/cloud-provider"
+	"k8s.io/component-base/cli/flag"
 	"k8s.io/component-base/logs"
+	"k8s.io/klog/v2"
 	"k8s.io/kubernetes/cmd/cloud-controller-manager/app"
 
 	_ "k8s.io/component-base/metrics/prometheus/clientgo" // load all the prometheus client-go plugins
 	_ "k8s.io/component-base/metrics/prometheus/version"  // for version metric registration
 )
 
+const (
+	cloudProviderName = "SampleCloudProviderName"
+	cloudConfigFile   = "SampleCloudProviderConfigFile"
+)
+
 func main() {
 	rand.Seed(time.Now().UnixNano())
 
-	command := app.NewCloudControllerManagerCommand()
+	cloud, err := cloudprovider.InitCloudProvider(cloudProviderName, cloudConfigFile)
```

Comment:
I think we'd still want to call this after flag parsing since a provider will likely depend on flags for the cloud config file path.

Reply:
In fact, it would be good to update this reference to parse the cloud config file from the flag option.

Reply:
I think it might be nice to show a split example. Have 1 parameter (provider name) hard coded to demonstrate they can get rid of these flag options and the other parameter (cloud config file) which as Andrew suggests demonstrates how to consume flag options.